### PR TITLE
Add cancellation token support

### PR DESCRIPTION
**Pull request by JoeyMckenzie**
Addresses issue #20, with accompanying updates to the sample project and additional unit tests.

### CODE REVIEW

File: sample/Sample.FunctionalTests/AuthorEndpoints/CreateEndpoint.cs [hunk 1]
```diff
@@ -44,5 +46,25 @@ public async Task CreatesANewAuthor()
             Assert.Equal(result.PluralsightUrl, newAuthor.PluralsightUrl);
             Assert.Equal(result.TwitterAlias, newAuthor.TwitterAlias);
         }
+
+        [Fact]
+        public async Task GivenLongRunningCreateRequest_WhenTokenSourceCallsForCancellation_RequestIsTermainated()
```
Comment (ardalis) [inline sample/Sample.FunctionalTests/AuthorEndpoints/CreateEndpoint.cs:51]:
```suggestion
        public async Task GivenLongRunningCreateRequest_WhenTokenSourceCallsForCancellation_RequestIsTerminated()
```

File: sample/Sample.FunctionalTests/AuthorEndpoints/GetEndpoint.cs [hunk 2]
```diff
@@ -34,5 +36,19 @@ public async Task ReturnsAuthorById()
             Assert.Equal(firstAuthor.PluralsightUrl, result.PluralsightUrl);
             Assert.Equal(firstAuthor.TwitterAlias, result.TwitterAlias);
         }
+
+        [Fact]
+        public async Task GivenLongRunningGetRequest_WhenTokenSourceCallsForCancellation_RequestIsTermainated()
```
Comment (ardalis) [inline sample/Sample.FunctionalTests/AuthorEndpoints/GetEndpoint.cs:41]:
```suggestion
        public async Task GivenLongRunningGetRequest_WhenTokenSourceCallsForCancellation_RequestIsTerminated()
```

File: sample/Sample.FunctionalTests/AuthorEndpoints/ListEndpoint.cs [hunk 3]
```diff
@@ -30,5 +32,18 @@ public async Task ReturnsTwoGivenTwoAuthors()
             Assert.NotNull(result);
             Assert.Equal(SeedData.Authors().Count(), result.Count());
         }
+
+        [Fact]
+        public async Task GivenLongRunningListRequest_WhenTokenSourceCallsForCancellation_RequestIsTermainated()
```
Comment (ardalis) [inline sample/Sample.FunctionalTests/AuthorEndpoints/ListEndpoint.cs:37]:
```suggestion
        public async Task GivenLongRunningListRequest_WhenTokenSourceCallsForCancellation_RequestIsTerminated()
```

File: sample/Sample.FunctionalTests/AuthorEndpoints/ListPaginatedEndpoint.cs [hunk 4]
```diff
@@ -28,7 +30,20 @@ public async Task Page1PerPage1_ShouldReturnFirstAuthor()
             var result = JsonConvert.DeserializeObject<IEnumerable<Author>>(stringResponse);
 
             Assert.NotNull(result);
-            Assert.Equal(1, result.Count());
+            Assert.Single(result);
+        }
+
+        [Fact]
+        public async Task GivenLongRunningPaginatedListRequest_WhenTokenSourceCallsForCancellation_RequestIsTermainated()
```
Comment (ardalis) [inline sample/Sample.FunctionalTests/AuthorEndpoints/ListPaginatedEndpoint.cs:37]:
```suggestion
        public async Task GivenLongRunningPaginatedListRequest_WhenTokenSourceCallsForCancellation_RequestIsTerminated()
```

File: sample/Sample.FunctionalTests/AuthorEndpoints/UpdateEndpoint.cs [hunk 5]
```diff
@@ -44,5 +46,24 @@ public async Task UpdateAnExistingAuthor()
             Assert.Equal(result.PluralsightUrl, authorPreUpdate.PluralsightUrl);
             Assert.Equal(result.TwitterAlias, authorPreUpdate.TwitterAlias);
         }
+
+        [Fact]
+        public async Task GivenLongRunningUpdateRequest_WhenTokenSourceCallsForCancellation_RequestIsTermainated()
```
Comment (ardalis) [inline sample/Sample.FunctionalTests/AuthorEndpoints/UpdateEndpoint.cs:51]:
```suggestion
        public async Task GivenLongRunningUpdateRequest_WhenTokenSourceCallsForCancellation_RequestIsTerminated()
```

**Comment by JoeyMckenzie** [comment]
Good catch, updated with latest commit.

**Pull request merged**